### PR TITLE
Extensions by classes or by jar

### PR DESCRIPTION
**Pull request by srose**

### CODE REVIEW

File: start.java [hunk 1]
```diff
@@ -64,6 +70,7 @@ public static void main(String[] args) {
             System.out.printf("  %s %s%n", "java start.java --https", "# Start Keycloak Environment with https");
             System.out.printf("  %s %s%n", "java start.java --https --database=postgres", "# Start Keycloak Environment with PostgreSQL database");
             System.out.printf("  %s %s%n", "java start.java --https --openldap --database=postgres", "# Start Keycloak Environment with PostgreSQL database and OpenLDAP");
+            System.out.printf("  %s %s%n", "java start.java --extensions=classes", "# Start Keycloak with extensions mounted from classes folder");
```
Comment (thomasdarimont) [inline start.java:73]: "Start Keycloak with extensions mounted from classes folder. Use --extensions=jar to mount the jar file into the container"

File: start.java [hunk 2]
```diff
@@ -53,6 +58,7 @@ public static void main(String[] args) {
             System.out.printf("  %s: %s%n", HTTPS_OPT, "enables HTTPS support. (Optional) Implies --http. If not provided, plain HTTP is used");
             System.out.printf("  %s: %s%n", OPENLDAP_OPT, "enables OpenLDAP support. (Optional)");
             System.out.printf("  %s: %s%n", POSTGRES_OPT, "enables postgrase database support. (Optional) If no other database is provided, H2 database is used");
+            System.out.printf("  %s: %s%n", EXTENSIONS_OPT, "enables deployemnt style, choose hot (default) or package depending on your needs");
```
Comment (thomasdarimont) [inline start.java:61]: "enables deployemnt style, choose hot (default) or package depending on your needs" -> This sound strange: we have a dynamic deployment based on "classes" and static based on "jar".

File: start.java [hunk 3]
```diff
@@ -95,6 +102,17 @@ public static void main(String[] args) {
             commandLine.add("deployments/local/dev/docker-compose-openldap.yml");
         }
 
+        if (EXTENSIONS_OPT_CLASSES.equals(deployment)) {
+            commandLine.add("--file");
+            commandLine.add("deployments/local/dev/docker-compose-extensionsclasses.yml");
```
Comment (thomasdarimont) [inline start.java:107]: docker-compose-extension-classes.yml

File: start.java [hunk 4]
```diff
@@ -95,6 +102,17 @@ public static void main(String[] args) {
...
+        } else if(EXTENSIONS_OPT_JAR.equals(deployment)) {
+            commandLine.add("--file");
+            commandLine.add("deployments/local/dev/docker-compose-extensionsjar.yml");
```
Comment (thomasdarimont) [inline start.java:110]: docker-compose-extension-jar.yml

**Review by thomasdarimont — CHANGES_REQUESTED** [review]

File: start.java [hunk 5]
```diff
@@ -95,6 +102,17 @@ public static void main(String[] args) {
...
         }
 
+        if (EXTENSIONS_OPT_CLASSES.equals(deployment)) {
+            commandLine.add("--file");
+            commandLine.add("deployments/local/dev/docker-compose-extensions-classes.yml");
+        } else if(EXTENSIONS_OPT_JAR.equals(deployment)) {
+            commandLine.add("--file");
+            commandLine.add("deployments/local/dev/docker-compose-extensions-jar.yml");
+        } else {
+            System.out.printf("Unkown deployment %s%n", deployment);
```
Comment (thomasdarimont) [inline start.java:112]: Should we exit here?

**Review by thomasdarimont — CHANGES_REQUESTED** [review]

File: start.java (hunk 5 above)
Comment (srose) [reply start.java:112]: Good question, the text should contain a warning, that no extensions are mounted and just continue? I am also fine with exit.

Comment (thomasdarimont) [reply start.java:112]: System.err.printf(...)

then I'd opt for System.exit(-1);

Comment (srose) [reply start.java:112]: :+1:

**Review by thomasdarimont — APPROVED** [review]

**Comment by thomasdarimont** [comment]
Thanks

**Pull request merged**